### PR TITLE
Redux in components

### PR DESCRIPTION
**Pull request by Kaiserabbas**
## Use of Redux in React Components 
- Wraped your <App> with the <Provider> component from react-redux
- Passed configured store into the <Provider>
- Created an initialState variable for booksSlice, which will include the following:
// Initial state:
[
  {
    id: 1,
    title: 'The Hunger Games',
    category: 'Action',
    author: 'Suzanne Collins',
    progress: 74,
    state: 'Chapter 17',
  },
  ......
  ]
- Displayed books, received from the slice, in a reusable component
- Dispatched actions using use dispatch
- Added a <Button> component, which includes:
--An event handler that adds a book to the books array (with attributes id, title, and author)
- Added a <Button> component, which includes:
-- An event handler that removes a book from the books array (by id)

### CODE REVIEW

**Review by Ibizugbe — COMMENTED** [review]
Hi @Kaiserabbas 👋🏼, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have the microverse readme template properly used. kindly do well to make use of the microverse readme template and ensure all sections are filled according to the instruction👍🏼.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
+import './App.css';
+import Nav from './components/Nav';
+import store from './redux/store';
 import Books from './components/Books';
 import Categories from './components/Categories';
-import Nav from './components/Nav';
 
 function App() {
   return (
     <Provider store={store}>
```
Comment (V-Blaze) [inline src/App.js]: -  You have done great so far but there are few changes you need to make before moving to the next milestone. :+1: :+1: 

- [ ] Looking at the screenshot below, from `Book.js`, it seems you also wrap the component with the redux provider, that is not needed as you already did that in the App.js file. I'll suggest you remove that. :+1: :+1: 

- - Screenshot for reference:
 
Screenshot from 2023-08-28 11-14-40


- [ ] Also following the best convention, I'll suggest you remove the `<Provider>` from App.js and move it to the index.js file. Like this:

```sh
import React from 'react';
import * as ReactDOM from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import reportWebVitals from './reportWebVitals';

// Redux
import { Provider } from 'react-redux';
import store from './redux/store';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <BrowserRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </BrowserRouter>
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();

```

File: src/components/RemoveButton.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      Remove <span className="vertical-line">|</span>
+    </button>
+  );
+};
+
+RemoveButton.propTypes = {
+  bookId: PropTypes.number.isRequired,
+};
+
+export default RemoveButton;
```
Comment (V-Blaze) [inline src/components/RemoveButton.js:26]: - [ ] Looking at your `.eslintrc.json`, it seems you are using the wrong or modified code, The content of the file is different from the one in the official Microverse Linters repository and it affects the Linters check, some errors in the code are not picked up by linters. I'll suggest you go to this repository [ https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json ] and copy the correct `.eslintrc.json` file content. Make sure to run the linters check after that and fix any error that your code has. NB: It is against Microverse rules to Modify any Linters code, the linters are there to make sure your code follows the best standard. :smile: :wave: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Kaiserabbas 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Books are displayed from redux state✔
- Adding and deleting works great👋 
-  Redux store passed to Provider component correctly ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Kaiserabbas,

✔️ Use of local state for book form component

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**